upload/upload.go

```
@@ -84,6 +62,80 @@ func DeleteAll(driver Driver, recipeId int64) error {
 	return err
 }
 
+// OptimizeImages regenerates all images and thumbnails from the currently saved originals
+func OptimizeImages(driver Driver, recipeId int64) error {
+	recipePath := getDirPathForImage(recipeId)
+	imagePaths, err := driver.List(recipePath)
+	if err != nil {
+		return err
+	}
+
+	for _, imagePath := range imagePaths {
+		file, err := driver.Open(imagePath)
+		if err != nil {
+			// TODO: Log and move on?
+			return err
+		}
+
+		stat, err := file.Stat()
+		if err != nil {
+			// TODO: Log and move on?
+			return err
+		}
+		log.Debug().Msg(stat.Name())
+
+		origData, err := ioutil.ReadAll(file)
```

Error is never checked